src/base/bittorrent/session.cpp

```diff
@@ -4064,6 +4069,35 @@ void Session::handleMetadataReceivedAlert(const lt::metadata_received_alert *p)
     }
 }
 
+void Session::handleTorrentPausedAlert(const libtorrent::torrent_paused_alert *p)
+{
+    const InfoHash hash {p->handle.info_hash()};
+
+    if (m_loadedMetadata.contains(hash)) {
+        // Adding preloaded torrent
+        m_loadedMetadata.remove(hash);
+        lt::torrent_handle handle = m_nativeSession->find_torrent(hash);
```

Comment:
seems you can get `torrent_handle` from `p` instead.

Reply:
Sure.